htdocs/core/lib/functions.lib.php

```diff
@@ -7920,3 +7921,242 @@ function roundUpToNextMultiple($n, $x = 5)
 {
 	return (ceil($n)%$x === 0) ? ceil($n) : round(($n+$x/2)/$x)*$x;
 }
+
+/**
+ * @param string $label     label of badge no html : use in alt attribute for accessibility
+ * @param string $html      optional : label of badge with html
+ * @param string $type      type of badge : Primary Secondary Success Danger Warning Info Light Dark status0 status1 status2 status3 status4 status5 status6 status7 status8 status9
+ * @param string $mode      default '' , pill, dot
+ * @param string $url       the url for link
+ * @param array $params     various params for future : recommended rather than adding more fuction arguments
+ * @return string   html badge
+ */
+function dol_get_badge($label, $html = '', $type = 'primary', $mode = '', $url = '', $params = array())
+{
+    
+    $attr=array(
+        'class'=>'badge'.(!empty($mode)?' badge-'.$mode:'').(!empty($type)?' badge-'.$type:'')
+    );
+    
+    if(empty($html)){
+        $html = $label;
+    }
+    
+    if(!empty($url)){
+        $attr['href'] = $url;
+    }
+    
+    if($mode==='dot')
+    {
+        $attr['class'].= ' classfortooltip';
+        $attr['title'] = $html;
+        $attr['aria-label'] = $label;
+        $html='';
+    }
+    
+    // Override attr
+    if(!empty($params['attr']) && is_array($params['attr'])){
+        foreach($params['attr']as $key => $value){
+            $attr[$key] = $value;
+        }
+    }
+
+    // TODO: add hook
+    
+    // escape all attribute
+    $attr = array_map('dol_escape_htmltag', $attr);
+    
+    $TCompiledAttr = array();
+    foreach($attr as $key => $value){
+        $TCompiledAttr[] = $key.'="'.$value.'"';
+    }
+    
+    $compiledAttributes = !empty($TCompiledAttr)?implode(' ',$TCompiledAttr):'';
+    
+    $tag = !empty($url)?'a':'span';
+
+    return '<'.$tag.' '.$compiledAttributes.'>'.$html.'</'.$tag.'>';
+}
+
+
+/**
+ * @param string $label     label of badge no html : use in alt attribute for accessibility
+ * @param string $html      optional : label of badge with html
+ * @param string $type      type of badge : Primary Secondary Success Danger Warning Info Light Dark status0 status1 status2 status3 status4 status5 status6 status7 status8 status9
+ * @param string $mode      default '' , pill, dot
+ * @param string $url       the url for link
+ * @param array $params     various params for future : recommended rather than adding more function arguments * 
+ * @return void print html bagde
+ */
+function dol_print_badge($label, $html = '', $type = 'primary', $mode = '', $url = '', $params = array())
+{
+    print dol_get_badge($label, $html, $type, $mode, $url, $params);
```

Comment:
Do we really need this function ? 
The caller can just do 
print dol_get_badge(
instead of 
dol_get_badge(
It looks better.

Reply:
it's to be like dol_print_date or all dol_print_* functions
in fact it's for developers IDE : when starting typing "dol_print_" your IDE show you all possibilities

Reply:
Forget the old naming, we will have to rename one day all function to be camel case. Let's start with new one...

Reply:
done